KubeArmor/types/types.go

```diff
@@ -424,6 +424,34 @@ type CapabilitiesType struct {
 	Action   string   `json:"action,omitempty"`
 }
 
+type SyscallFromSourceType struct {
+	Path string `json:"path,omitempty"`
+}
+
+type SyscallMatchType struct {
+	Syscalls   []string                `json:"syscall,omitempty"`
+	FromSource []SyscallFromSourceType `json:"fromSource,omitempty"`
+}
+
+type SyscallMatchPathType struct {
+	Path       string                  `json:"path,omitempty"`
+	Syscalls   []string                `json:"syscall,omitempty"`
+	FromSource []SyscallFromSourceType `json:"fromSource,omitempty"`
+}
+
+type SyscallMatchDirectoriesType struct {
+	Dir        string                  `json:"dir,omitempty"`
+	Recursive  bool                    `json:"recursive,omitempty"`
+	Syscalls   []string                `json:"syscall,omitempty"`
+	FromSource []SyscallFromSourceType `json:"fromSource,omitempty"`
+}
```

Comment:
Lets not have Paths and Directories separately, like we have for policy rules today.
If a path ends with `/` then it is a directory.